### PR TITLE
C209 Tag SW version 2.0 

### PR DESCRIPTION
**Pull request by jakkra**
Some of the major new features:
- Bluetooth compliant Direction Finding using periodic advertisements
- AT commands over Nordic NUS service

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,52 +1,49 @@
...
+For more detailed instructions go to the offical [nRF Connect Getting Started Guide](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/getting_started.html)
+
+## Download nRF Connect and Toolchain
+- Open nRF Connect for Desktop and open the Toolchain Manager.
+- Download nRF Connect SDK v2.1.0.
+
+### Building with nRF Connect VSCode plugin
+- In Visual Studio Code go to marketplace and search for "nrf connect".
+- Install the nRF Connect Extension Pack (this will install all required extensions at once).
+- Click on the nRF Connect extention and select "Add an existing application"
```
Comment (perssonmagnus) [inline README.md]: Typo, extention -> extension

File: scripts/flash_tags.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+from multiprocessing import Process
+
+
+def run_command(cmd):
+    subprocess.run(cmd)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Tool to flash a list of COM ports using nRF DFU bootloader (nrfutil tool)."
```
Comment (perssonmagnus) [inline scripts/flash_tags.py]: Do we need a comment about nrfutil in the PATH?

**Review by perssonmagnus — COMMENTED** [review]
Looks good to me! Only 2 very minor comments.

File: scripts/flash_tags.py (hunk 2 above)
Comment (jakkra) [reply scripts/flash_tags.py]: Added, also added some more documentation for those helper scripts in 4d23d207e3c565028417b6fa8157bddd8bedd4b0.

**Review by perssonmagnus — APPROVED** [review]
Looks good to me!

File: src/bt_adv.c [hunk 3]
```diff
@@ -20,59 +20,211 @@
...
+{
+    minAdvInterval = min_int / 1.25;
+    maxAdvInterval = max_int / 1.25;
+    advRunning = false;
+
+    memcpy((uint8_t *)&ad[2].data[ADV_DATA_OFFSET_NAMESPACE], namespace, EDDYSTONE_NAMESPACE_LENGFTH);
+    memcpy((uint8_t *)&ad[2].data[ADV_DATA_OFFSET_INSTANCE], instance_id, EDDYSTONE_INSTANCE_ID_LEN);
+    memcpy((uint8_t *)&ad[2].data[ADV_DATA_OFFSET_TX_POWER], &txPower, sizeof(txPower));
+
+    int err = bt_le_ext_adv_create(&param, NULL, &adv_set);
```
Comment (plerup) [inline src/bt_adv.c:114]: Missing LOG string here?

File: src/bt_adv.c [hunk 4]
```diff
@@ -20,59 +20,211 @@
...
-void btAdvStop(void) {
-    bt_le_adv_stop();
+void btAdvStop(void)
+{
+    if (!advRunning) {
+        LOG_WRN("Periodic adv. already stopped");
+        return;
+    }
+    __ASSERT_NO_MSG(0 == bt_le_per_adv_stop(adv_set));
+    __ASSERT_NO_MSG(0 == bt_le_ext_adv_stop(adv_set));
```
Comment (plerup) [inline src/bt_adv.c]: Why twice?

File: src/main.c [hunk 5]
```diff
@@ -188,4 +266,53 @@ static void setTxPower(uint8_t handleType, uint16_t handle, int8_t txPwrLvl)
...
+#if defined(CONFIG_BT_NUS)
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+    char addr[BT_ADDR_LE_STR_LEN];
+
+    if (err) {
+        LOG_ERR("Connection failed (err %u)", err);
+        return;
+    }
+    current_conn = bt_conn_ref(conn);;
```
Comment (plerup) [inline src/main.c]: Double ;

**Review by plerup — APPROVED** [review]
LGTM
Just some minor comments

**Pull request merged**